src/state.proto

```diff
@@ -0,0 +1,8 @@
+syntax = "proto3";
+
+message TracerState {
```

Comment:
this really should be brought in through a submodule or something, but let's leave it this way for now.